### PR TITLE
Issue #991: Add context with DataVersion to ContentBlockSource

### PR DESCRIPTION
**Pull request by Vinai**
This is a preparatory refactoring to add a content block api data_version parameter.

### CODE REVIEW

File: src/Import/ContentBlock/ContentBlockSnippetRenderer.php [hunk 1]
```diff
@@ -15,18 +15,11 @@ class ContentBlockSnippetRenderer implements SnippetRenderer
      * @var SnippetKeyGeneratorLocator
      */
     private $snippetKeyGeneratorLocator;
-
-    /**
-     * @var ContextBuilder
```
Comment (tim-bezhashvyly) [inline src/Import/ContentBlock/ContentBlockSnippetRenderer.php]: Probably also remove an import?

File: src/Import/ContentBlock/ContentBlockSnippetRenderer.php [hunk 2]
```diff
@@ -15,18 +15,11 @@ class ContentBlockSnippetRenderer implements SnippetRenderer
...
-     */
-    private $contextBuilder;
-
+    
     public function __construct(
-        SnippetKeyGeneratorLocator $snippetKeyGeneratorLocator,
-        ContextBuilder $contextBuilder
+        SnippetKeyGeneratorLocator $snippetKeyGeneratorLocator
     ) {
```
Comment (tim-bezhashvyly) [inline src/Import/ContentBlock/ContentBlockSnippetRenderer.php]: Most likely will fit into a single line now.

File: src/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandler.php [hunk 3]
```diff
@@ -24,9 +27,24 @@ class ContentBlocksApiV1PutRequestHandler extends ApiRequestHandler
...
+    /**
+     * @var DataPoolReader
+     */
+    private $dataPoolReader;
+
+    public function __construct(
+        CommandQueue $commandQueue,
+        ContextBuilder $contextBuilder,
+        DataPoolReader $dataPoolReader
+    ) {
```
Comment (tim-bezhashvyly) [inline src/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandler.php]: Wouldn't it fit into a single line?

File: tests/Unit/Suites/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandlerTest.php [hunk 4]
```diff
@@ -41,10 +47,36 @@ class ContentBlocksApiV1PutRequestHandlerTest extends \PHPUnit_Framework_TestCas
...
-        $this->requestHandler = new ContentBlocksApiV1PutRequestHandler($this->mockCommandQueue);
+        $this->stubContextBuilder = $this->createMock(ContextBuilder::class);
+        $this->stubContextBuilder->method('createContext')->willReturnCallback(function (array $parts) {
+            $stubContext = $this->getMockBuilder(Context::class)
+                ->disableOriginalConstructor()
+                ->setMethods(array_merge(get_class_methods(Context::class), ['debug']))
+                ->getMock();
+            $stubContext->method('debug')->willReturn($parts);
+            return $stubContext;
+
```
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandlerTest.php]: Superfluous newline.

File: src/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandler.php (hunk 3 above)
Comment (Vinai) [inline src/Import/ContentBlock/RestApi/ContentBlocksApiV1PutRequestHandler.php]: No, unfortunately not.

**Comment by Vinai** [comment]
Ready for next review.

**Pull request merged**